File: backend/app/schemas/trade.py
# backend/app/schemas/trade.py

from datetime import date
from typing import Optional
from app.schemas.base import BaseResponse
from app.enums.trade import TradeStatus, ExitReason


class SecurityInfo(BaseResponse):
    id: int
    ticker: str
    display_name: str
    sector: Optional[str] = None
    industry: Optional[str] = None


class TradeResponse(BaseResponse):
    id: int
    security: SecurityInfo
    status: TradeStatus
    entry_date: date
    fill_price: Optional[float] = None
    fill_quantity: Optional[int] = None
    timeout_date: date
    exit_date: Optional[date] = None
    exit_reason: Optional[ExitReason] = None
    state: dict
    invested_value: Optional[float] = None
    pnl: Optional[float] = None
    pnl_pct: Optional[float] = None

File: frontend/libraries/types/trade.ts
export type TradeStatus = "open" | "pending" | "closed"
export type ExitReason = "atr_stop" | "timeout" | "manual"

export interface Security {
    id: number
    ticker: string
    display_name: string
    sector: string | null
    industry: string | null
}

export interface Trade {
    id: number
    security: Security
    status: TradeStatus
    entry_date: string
    fill_price: number | null
    fill_quantity: number | null
    timeout_date: string
    exit_date: string | null
    exit_reason: ExitReason | null
    state: Record<string, unknown>
    invested_value: number | null
    pnl: number | null
    pnl_pct: number | null
}
